staging/src/k8s.io/component-base/metrics/testutil/testutil.go

```diff
@@ -68,3 +69,21 @@ func NewFakeKubeRegistry(ver string) metrics.KubeRegistry {
 
 	return metrics.NewKubeRegistry()
 }
+
+func GetConcurrencyDataOfPriorityLevel(response, metricName, labelPriorityLevel string) (map[string]int, error) {
+	concurrency := make(map[string]int)
+
+	samples, err := ExtractMetricSamples(response)
+
+	if err != nil {
+		return nil, err
+	}
+	for _, metric := range samples {
+		switch name := string(metric.Metric[model.MetricNameLabel]); name {
+		case metricName:
+			concurrency[string(metric.Metric[model.LabelName(labelPriorityLevel)])] = int(metric.Value)
+		}
+	}
+
+	return concurrency, nil
+}
```

Comment:
This is a very specific function for (what it sounds like) a very specific metric.. why are you extracting this to testutils?

Reply:
1. remove dependency
2. I don't find that similar function in staging/src/k8s.io/component-base/metrics/testutil/
the similar function is 
```
// GetMetricValuesForLabel returns value of metric for a given dimension
func GetMetricValuesForLabel(ms Metrics, metricName, label string) map[string]int64 {
	samples, found := ms[metricName]
	result := make(map[string]int64, len(samples))
	if !found {
		return result
	}
	for _, sample := range samples {
		count := int64(sample.Value)
		dimensionName := string(sample.Metric[model.LabelName(label)])
		result[dimensionName] = count
	}
	return result
}
```

3. I guess that we can have this function in testutil for (input ,metricName, label ) to extract value

```
// ExtractMetricSamples parses the prometheus metric samples from the input string.
func ExtractMetricSamples(metricsBlob string) ([]*model.Sample, error) {
	dec := expfmt.NewDecoder(strings.NewReader(metricsBlob), expfmt.FmtText)
	decoder := expfmt.SampleDecoder{
		Dec:  dec,
		Opts: &expfmt.DecodeOptions{},
	}

	var samples []*model.Sample
	for {
		var v model.Vector
		if err := decoder.Decode(&v); err != nil {
			if err == io.EOF {
				// Expected loop termination condition.
				return samples, nil
			}
			return nil, err
		}
		samples = append(samples, v...)
	}
}
```
Or do you have other good ideas？

Reply:
Yes, I think I prefer just extracting out `ExtractMetricSamples` into the metrics directory. I believe you will have to create a wrapper around the `prometheus.Sample` struct if we don't have one for it already. (@RainbowMango probably knows better than I at this point)

Reply:
thanks . i will think more about it.

Reply:
/cc @logicalhan 
how about GetMetricValueFromSample , it is more common.